src/flint/FlintTypes.jl

```diff
@@ -5948,7 +5978,12 @@ mutable struct FqMatrix <: MatElem{FqFieldElem}
   base_ring::FqField
   view_parent
 
-  # used by windows, not finalised!!
+  # MatElem interface
+  function FqMatrix(R::FqField, r::Int, c::Int)
+    return FqMatrix(r, c, R)
```

Comment:
For `FqMatrix` we already have `FqMatrix(r, c, R)` but I don't see a downside with having both that and `FqMatrix(R, r, c)`. Down the road we could think about abolishing the former in a breaking release, but there is no strong need either.

(One could also argue that we should just use the `(r, c, R)` order everywhere. Would be also OK to me. I used the former because (a) the old PR used it and (b) it matches the order in `matrix(R, r, c, ...)`, `zero_matrix`, and some other places

I think @thofma added the `FqMatrix` constructor here, so perhaps he had a deeper reason for the different order that we should heed?

Reply:
The reason is that I copy & pasted it from some other matrix constructor in Nemo. There is no deeper reason. It is not part of the API, so can be changed/augmented/removed.